Shopify/hydrogen
# Analytics feedbacks

### WHY are these changes introduced?

* Fixes the case when we are using the customer privacy api without the banner (for integrating with 3P consent management tools), the `<Analytics.Provider>` will not send any events.
* Make sure `consent.checkoutDomain` and `consent.storefrontAccessToken` is required
* Fixes https://github.com/Shopify/hydrogen/issues/2026
* Make sure customer privacy is not using legacy functions

Introduce `onReady` callback for `useCustomerPrivacy` hook.

### WHAT is this pull request doing?





### HOW to test your changes?

Use `analytics` example:

1. Test errors
    * Remove `consent.checkoutDomain` from root loader --> site should generate error about missing props
    * Remove `consent.storefrontAccessToken` from root loader --> site should generate error about missing props
2. Test working analytics
    * Add `consent.withPrivacybanner: false` in the root loader --> You should see `CustomAnalytics` logging page views in the console log

#### Post-merge steps



#### Checklist
